lib/wallet/wallet.js

```diff
@@ -3406,32 +3406,170 @@ class Wallet extends EventEmitter {
   }
 
   /**
-   * Create and finalize a renewal
+  * @typedef {Object} Renewal
+  * @property {Output} output
+  * @property {Outpoint} outpoint
+  */
+
+  /**
+  * process a renewal
+  * @private
+  * @param {String} name
+  * @param {Number} acct
+  * @returns {Promise<Renewal>}
+  */
+
+  async processRenewal(name, acct) {
+    assert(typeof name === 'string');
+
+    if (!rules.verifyName(name))
+      throw new Error(`Invalid name: "${name}".`);
+
+    const rawName = Buffer.from(name, 'ascii');
+    const nameHash = rules.hashName(rawName);
+    const ns = await this.getNameState(nameHash);
+    const height = this.wdb.height + 1;
+    const network = this.network;
+
+    if (!ns)
+      throw new Error(`Auction not found: "${name}".`);
+
+    const { hash, index } = ns.owner;
+    const coin = await this.getCoin(hash, index);
+
+    if (!coin)
+      throw new Error(`Wallet does not own: "${name}".`);
+
+    if (ns.isExpired(height, network))
+      throw new Error(`Name has expired: "${name}"!`);
+
+    // Is local?
+    if (coin.height < ns.height)
+      throw new Error(`Wallet does not own: "${name}".`);
+
+    if (acct != null && !await this.txdb.hasCoinByAccount(acct, hash, index))
+      throw new Error(`Account does not own: "${name}".`);
+
+    const state = ns.state(height, network);
+
+    if (state !== states.CLOSED)
+      throw new Error(`Auction is not yet closed: "${name}".`);
+
+    if (!coin.covenant.isRegister()
+        && !coin.covenant.isUpdate()
+        && !coin.covenant.isRenew()
+        && !coin.covenant.isFinalize()) {
+      throw new Error(`Name must be registered: "${name}".`);
+    }
+
+    if (height < ns.renewal + network.names.treeInterval)
+      throw new Error(`Must wait to renew: "${name}".`);
+
+    const output = new Output();
+    output.address = coin.address;
+    output.value = coin.value;
+    output.covenant.type = types.RENEW;
+    output.covenant.pushHash(nameHash);
+    output.covenant.pushU32(ns.height);
+    output.covenant.pushHash(await this.wdb.getRenewalBlock());
+
+    return { output, outpoint: ns.owner };
```

Comment:
Can you refactor makeRenewal so that it uses this helper code? This seems exactly duplicated except a few lines at the bottom so we should be careful it doesn't become stale

Reply:
atm makeRenewal is dead code, no point keeping it and there is no harm removing it we rely on makeBatchRenew to perform same action. I did remove it.